blitbuffer.c

```diff
@@ -145,13 +145,13 @@ static const char*
 #define BB_GET_PIXEL(bb, rotation, COLOR, x, y, pptr) \
 ({ \
     if (rotation == 0) { \
-        *pptr = (COLOR*)(bb->data + y * bb->stride) + x; \
+        *pptr = (COLOR*)(bb->data + (y) * bb->stride) + (x); \
     } else if (rotation == 1) { \
-        *pptr = (COLOR*)(bb->data + x * bb->stride) + bb->w - y - 1; \
+        *pptr = (COLOR*)(bb->data + (x) * bb->stride) + bb->w - (y) - 1; \
     } else if (rotation == 2) { \
-        *pptr = (COLOR*)(bb->data + (bb->h - y - 1) * bb->stride) + bb->w - x - 1; \
+        *pptr = (COLOR*)(bb->data + (bb->h - (y) - 1) * bb->stride) + bb->w - (x) - 1; \
     } else if (rotation == 3) { \
-        *pptr = (COLOR*)(bb->data + (bb->h - x - 1) * bb->stride) + y; \
+        *pptr = (COLOR*)(bb->data + (bb->h - (x) - 1) * bb->stride) + (y); \
```

Comment:
Why not fix the other arguments too?

Reply:
Not touching other people stuff other than what they allowed me to :) https://github.com/koreader/koreader/issues/10193#issuecomment-1741879785